src/main/java/org/cloudfoundry/promregator/scanner/ReactiveAppInstanceScanner.java

```diff
@@ -263,60 +369,71 @@ public void setNumberOfInstances(int numberOfInstances) {
 				return Mono.just(INVALID_SPACE_ID);
 			}
-			
+
 			SpaceResource spaceResource = resources.get(0);
 			return Mono.just(spaceResource.getMetadata().getId());
 		}).onErrorResume(e -> {
-			log.error(String.format("retrieving space id for org id '%s' and space name '%s' resulted in an exception", orgIdString, spaceNameString), e);
+			log.error(String.format("retrieving space id for org id '%s' and space name '%s' resulted in an exception",
+					orgIdString, spaceNameString), e);
 			return Mono.just(INVALID_SPACE_ID);
 		}).cache();
 
 	}
-	
+
 	private Mono<Map<String, SpaceApplicationSummary>> getSpaceSummary(String spaceIdString) {
-		return this.cfAccessor.retrieveSpaceSummary(spaceIdString)
-			.flatMap(response -> {
-				List<SpaceApplicationSummary> applications = response.getApplications();
-				if (applications == null) {
-					return Mono.just(INVALID_SUMMARY);
-				}
-				
-				Map<String, SpaceApplicationSummary> map = new HashMap<>(applications.size());
-				for (SpaceApplicationSummary sas : applications) {
-					map.put(sas.getName().toLowerCase(LOCALE_OF_LOWER_CASE_CONVERSION_FOR_IDENTIFIER_COMPARISON), sas);
-				}
-				
-				return Mono.just(map);
-			}).onErrorResume(e -> {
-				log.error(String.format("retrieving summary for space id '%s' resulted in an exception", spaceIdString), e);
+		return this.cfAccessor.retrieveSpaceSummary(spaceIdString).flatMap(response -> {
+			List<SpaceApplicationSummary> applications = response.getApplications();
+			if (applications == null) {
 				return Mono.just(INVALID_SUMMARY);
-			});
+			}
+
+			Map<String, SpaceApplicationSummary> map = new HashMap<>(applications.size());
+			for (SpaceApplicationSummary sas : applications) {
+				map.put(sas.getName().toLowerCase(LOCALE_OF_LOWER_CASE_CONVERSION_FOR_IDENTIFIER_COMPARISON), sas);
+			}
+
+			return Mono.just(map);
+		}).onErrorResume(e -> {
+			log.error(String.format("retrieving summary for space id '%s' resulted in an exception", spaceIdString), e);
+			return Mono.just(INVALID_SUMMARY);
+		});
 	}
-	
-	private String determineAccessURL(final String protocol, final List<String> urls, final List<Pattern> preferredRouteRegex, final String path) {
-		
-		final String url = determineApplicationRoute(urls, preferredRouteRegex);
+
+	private String formatAccessURL(final String protocol, final String url, final String path) {
 		final String applicationUrl = String.format("%s://%s", protocol, url);
-		log.debug (String.format("Using Application URL: '%s'", applicationUrl));
-		
+		log.debug(String.format("Using Application URL: '%s'", applicationUrl));
+
 		String applUrl = applicationUrl;
 		if (!applicationUrl.endsWith("/")) {
 			applUrl += '/';
 		}
-		
+
 		String internalPath = path;
 		while (internalPath.startsWith("/")) {
 			internalPath = internalPath.substring(1);
 		}
-		
+
 		return applUrl + internalPath;
 	}
 
+	private String formatInternalAccessURL(final String url, final String path, final int internalRoutePort,
```

Comment:
I find the name of `url` a little confusing. This is more something like `hostnameDomain` or so...

Reply:
Yup i have updated